lib/ReplaceOpenCLBuiltinPass.cpp

```diff
@@ -2028,6 +2039,10 @@ bool ReplaceOpenCLBuiltinPass::replaceVstore(Function &F) {
     // Avoid pointer casts. Instead generate the correct number of stores
     // and rely on drivers to coalesce appropriately.
     IRBuilder<> builder(CI);
+    if (clspv::PointersAre64Bit(*F.getParent())) {
```

Comment:
It seems like some of the functions are treating indices as 64-bit and some as 32-bit. Why aren't the consistently based on the size_t arguments? Here, for example, the offset parameter could be examined. If the primitive size is 64-bits then use 64-bits throughout.

Reply:
Generally I've tried to keep them all treating the indices as 64-bit where possible, but it does seem like there are a few inconsistent ones so I'll revisit those. `vstore_half` is a weird one because in the path where it uses an atomic_xor it uses the offset to create a mask for an i32 value, but I think it can be changed to avoid truncation

Reply:
I've updated these so it should be consistently using size_t now